### DISCUSSION TITLE
Debug odoo using vscode and develop environment

### DISCUSSION THREAD

**Discussion by rquevedo** [Tecnativa/doodba#429] [Q&A]
Hi folks,
When i try to run and debug python using vscode and develop environment this error shows in terminal:

`odoo_proxy_1                  | 2021/07/12 23:22:32 socat[292] E connect(5, AF=2 192.168.48.12:6899, 16): Connection refused`

Any ideas about what im doing wrong?

If i run odoo in standard mode(not debug) everything is fine.

**Comment by joao-p-marques** [comment]
Hi @rquevedo,

Can you provide some detailed steps to reproduce the problem?

**Reply by rquevedo** [reply]
Hi @joao-p-marques i just follow instructions of https://github.com/Tecnativa/doodba-copier-template to develop. Evrything was fine until i need to debug python code.Can you point me a config or file that play an important role in debugger attach process? Note that im very new using doodba as development environment even if i used since over a year as production environment in my vps.

**Reply by yajo** [reply]
You have docs in https://github.com/Tecnativa/doodba-copier-template/blob/main/docs/daily-usage.md#visual-studio-code-integration
